Simplified/Book/UI/NYPLBookCellDelegate+Audiobooks.m

```diff
@@ -216,6 +223,34 @@ - (void)scheduleTimerForAudiobook:(NYPLBook *)book
   }];
 }
 
+// Create a timer that saves the audiobook progress periodically when the app is inactive.
+// We do save the progress when app is being killed and applicationWillTerminate: is called,
+// but applicationWillTerminate: is not always called when users force quit the app.
+// This method is triggered when app resigns active.
+- (void)scheduleAudiobookProgressSavingTimer {
+  if (self.audiobookProgressSavingTimer) {
+    return;
+  }
+  
+  self.audiobookProgressSavingTimer = [NSTimer scheduledTimerWithTimeInterval:AudiobookProgressSavingInterval
+                                                                      repeats:YES
+                                                                        block:^(NSTimer *_Nonnull timer) {
+    [NYPLMainThreadRun asyncIfNeeded:^{
+      if ([[UIApplication sharedApplication] applicationState] == UIApplicationStateActive) {
+        [timer invalidate];
+        self.audiobookProgressSavingTimer = nil;
+        return;
+      }
+    }];
+    
+    if (self.manager && !self.manager.audiobook.player.isPlaying) {
+      return;
+    }
+    
+    [[NYPLBookRegistry sharedRegistry] save];
```

Comment:
is this what does the saving of the listening position?

Reply:
also, does this need to run in the main thread? it seems like this method saves to disk, so it can be expensive

Reply:
It now runs on a background thread